src/pages/index.tsx

```diff
@@ -1,22 +1,284 @@
-import { Button, Center } from '@chakra-ui/react';
-import { useEffect, useState } from 'react';
+import {
+  Heading,
+  Text,
+  Button,
+  Flex,
+  Image,
+  Box,
+  calc,
+  Fade,
+  Input,
+  HStack,
+} from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import { HamburgerIcon } from "@chakra-ui/icons";
 
-export default function Home() {
-  const [counter, setCounter] = useState(0);
+interface dataAPIType {
+  adult: boolean;
+  backdrop_path: string;
+  genre_ids: number[];
+  id: number;
+  original_language: string;
+  original_title: string;
+  overview: string;
+  popularity: number;
+  poster_path: string;
+  release_date: string;
+  title: string;
+  video: boolean;
+  vote_average: number;
+  vote_count: number;
+}
+
+const useScreenWidth = (): number | null => {
+  const [screenWidth, setScreenWidth] = useState<number | null>(null);
 
   useEffect(() => {
-    console.log('useEffect kepanggil');
+    const getWindowDimensions = (): number | null => {
+      return typeof window !== "undefined" ? window.innerWidth : null;
+    };
+
+    const handleResize = () => {
+      setScreenWidth(getWindowDimensions());
+    };
+
+    handleResize();
+
+    window.addEventListener("resize", handleResize);
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
   }, []);
 
+  return screenWidth;
+};
+
+const Fetch = ({ endpoint }: { endpoint: string }): dataAPIType[] => {
+  const [data, setData] = useState<dataAPIType[]>([]);
+  useEffect(() => {
+    const options = {
+      method: "GET",
+      url: `https://api.themoviedb.org/3/${endpoint}?language=en-US`,
+      headers: {
+        accept: "application/json",
+        Authorization:
+          "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJiMDAwODZmMjExMGJhODMwZDEwZDM3ZDhjYTM0ZGFhMiIsInN1YiI6IjY0OGM4OWFhMDc2Y2U4MDBhZDcyYjlmMyIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.GBTW1UV81mVUui2vRmHF7HL31dcH_VJPtDfTaI5eBII",
+      },
+    };
+
+    axios
+      .request(options)
+      .then(function (response) {
```

Comment:
Ada cara yang lebih clean, pake yg namanya async/await

Refer ke [callback hell](https://www.scaler.com/topics/callback-hell-in-javascript/) untuk tahu lebih lanjut